src/main/java/Scheduler.java

```diff
@@ -22,43 +24,121 @@ public Scheduler(ArrayList<Course> courses, ArrayList<Schedule> schedules) {
         this.schedules = schedules;
     }
 
+    public void addFilters(List<Filter> f) {
+        this.filters.addAll(f);
+    }
+
     /**
-     * Takes a list of course codes and outputs a schedule that takes the first lecture session and
-     * first tutorial session in each lecture.
+     * Scheduler used to create courses non recursively by permutation.
+     * Courses should be passed so that they are sorted in terms of priority.
+     */
+    public List<Schedule> permutationScheduler(List<Course> newCourses) {
+        if (newCourses.isEmpty()) {
+            return new ArrayList<>();
+        }
+        int numOfCourses = newCourses.size();
+
+        if (newCourses.size() == 1) {
+            Course newCourse = newCourses.get(0);
+            List<Schedule> newSchedules = populatePermutations(newCourse);
+            return newSchedules;
+        } else {
+            Course newCourse = newCourses.get(numOfCourses-1);
+            newCourses.remove(numOfCourses-1);
+            List<Schedule> savedSchedules = permutationScheduler(newCourses);
+            List<Schedule> newSchedules = new ArrayList<>();
+
+            for (Schedule schedule : savedSchedules) {
+                newSchedules.addAll(extendPermutations(newCourse, schedule));
+            }
+
+            savedSchedules.addAll(newSchedules);
+            return savedSchedules;
+        }
+    }
+
+    /**
+     * Takes a list of courses and outputs a schedule that takes the first lecture section and
+     * first tutorial section in each course. Mainly used for testing purposes.
      *
-     * @param courseCodes an ArrayList of course codes from which a schedule will be generated.
+     * @param courses an ArrayList of courses from which a schedule will be generated.
      */
-    public Schedule createBasicSchedule(ArrayList<String> courseCodes) {
+    public Schedule createBasicSchedule(ArrayList<Course> courses) {
         Schedule schedule = new Schedule();
-        Course newCourse;
-
-        for (String courseCode : courseCodes) {
-            try {
-                /**
-                 * For every course code, generate the course from CourseCreator, add first lec/tut
-                 * session to the lectures and tutorials within the schedule.
-                 */
-                newCourse = CourseCreator.generateCourse(courseCode, 'F');
-                System.out.println(newCourse);
-                if (!newCourse.getLectures().isEmpty()) {
-                    schedule.addLecture(newCourse.getLectures().get(0));
+
+        for (Course newCourse : courses) {
```

Comment:
I don't like the names newCourse and newCourses and all of that I think the naming should be simplified and less confusing